src/hooks/useBrandConfig.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { StyleSheet } from 'react-native';
+import { TextStyles, defaultTextStyles } from 'src/components/Text';
+import * as theme from '../theme';
+
+export interface BrandConfig {
+  styles: {
+    Text?: TextStyles | undefined;
+  };
+  theme: {};
+}
```

Comment:
I'm eager to find/name strategies for ensuring this type doesn't balloon with props that we later want to refactor/remove as a breaking change.  The only I can think of are:
- Generally keep this type super simple and be cautious about adding to it (maybe need a comment)
- I wonder if we could separate out different groups into different types that could be (optionally) decorated/merged into a generic type used by this provider/context/hook?  Can explain further offline

Reply:
Agree 💯 . The idea here (in contrast to some ways we have done this in the past), would be that `styles:` property is composed from collections of styles imported from each component.  

All the style typing information should live with each component and be collected together here.  Might even be good to create a helper to build this up in the future if it gets to be too big here.